chalk-solve/src/clauses.rs

```diff
@@ -161,7 +224,12 @@ fn program_clauses_that_could_match<I: Interner>(
 
             if trait_datum.is_non_enumerable_trait() || trait_datum.is_auto_trait() {
                 let self_ty = trait_ref.self_type_parameter(interner);
-                if self_ty.bound_var(interner).is_some()
+
+                if let TyData::Alias(AliasTy::Opaque(opaque_ty)) = self_ty.data(interner) {
```

Comment:
I *think* this probably wants to be the placeholder, and not the alias type, but we can leave it as "future work" for now